src/pool.c

```diff
@@ -288,6 +290,25 @@ hsk_pool_set_seeds(hsk_pool_t *pool, const char *seeds) {
   return true;
 }
 
+bool
+hsk_pool_set_agent(hsk_pool_t *pool, const char *user_agent) {
+  assert(pool);
+
+  if (!user_agent)
+    return true;
```

Comment:
did you mean to return false instead?

Reply:
No, if the user does not supply an extra string, then this function does nothing which counts as success. The default user agent has already been applied.

Reply:
As a user of this library if i call `hsk_pool_set_agent(pool, NULL)` and it returned true i expect my supplemental agent to be cleared but can't imagine a use case for this :D so that's fine

Reply:
Oh I see huh. I wonder actually if passing `""` would work. It's an empty string but wuoldn't be a NULL pointer ?

Reply:
yup passing empty string clears the agent/returns true that's why NULL shouldn't indicate success here but i'm just being nitpicky it is not really important